### DISCUSSION TITLE
Anflugkarten bulk download

### DISCUSSION THREAD

**Discussion by dorianiro** [mipastgt/AIPBrowserDE-Public#4] [Ideas]
Danke für das tolle Programm!
Ich wünsche mir die Möglichkeit, ALLE Anflugkaften mit den Rollkarten auf einen Rutsch runterladen zu können. Dann hat man sie als pdf auf dem tablet im flug in schnellem zugriff, wenn man zu einem nicht vorgeplanten Ausweichplatz muss.

**Comment by mipastgt** [comment]
ALLE Anflugkarten im wörtlichen Sinne wäre wohl etwas viel :-) aber in der nächsten Version 1.1.3 wird es eine Option geben, um eine Auswahl beliebiger Seiten der AIP als ein PDF-File exportieren zu können. (Momentan geht das schon für einzelne Flugplätze.) Das Feature ist bereits implementiert und ich schreibe gerade noch an der Dokumentation und bin noch am Testen.

Das ganze funktioniert über den Trip-Kit-Export über eine kleine Konfigurationsdatei. (Später soll das auch mal über das Benutzerinterface gehen.) Eine Beispieldatei sieht so aus:

```
{
    "version": "1.0.0",
    "name": "Hahnweide to Aachen",
    "description": "A trip from Hahnweide to Aachen via Speyer and Koblenz",
    "pages": {
        "pageFormat": "pdf",
		"references": [
			"Hahnweide EDST",
			"Speyer EDRY",
			"Koblenz-Winningen EDRK",
			"Aachen-Merzbrueck EDKA"
		]
    }
}
```
In diesem Beispiel werden ALLE Seiten der 4 genannten Flugplätze als eine einzige PDF-Datei ausgegeben. Die Reihenfolge entspricht der Reihenfolge in der Liste. Diese Liste lässt sich auch leicht und fehlerfrei erstellen, da der entsprechende Eintrag bei jeder Selektion im Inhaltsverzeichnis von AIPBrowserDE in die Zwischenablage kopiert wird. D.h. man braucht den Text nur noch über "Einfügen" im Text-Editor der Liste hinzufügen. Im PDF wird auch ein Inhaltsverzeichnis angelegt, so dass man die Seiten für einen Flugplatz zur besseren Übersicht einklappen kann. Ich hoffe dieses Feature erfüllt so Deine wichtigsten Anforderungen.

**Comment by dorianiro** [comment]
für die Anwendung Tripkit ist das natürlich toll. Wenn man aber ein paar Tage unterwegs ist und noch gar nicht weiß wo es einen hin verschlägt, hilft es nicht sehr, und dann will man ja auch nicht ein einziges PDF haben, sondern eins für jeden Platz.
Vor ein paar Monaten hatte sich ein Bekannter die Mühe gemacht, alle Anflug-Karten zwecks Geo-Referenzierung drunter zu laden, und mir doe pdfs gegeben. Leider immer nur die erste Seite, nicht auch die Rollkarten. Und natürlich veraltet alles...
da wäre es schon toll, tatsächlich alle Anflug-Karten auf einen Rutsch runterladen zu können.

Aktuell fliege ich deshalb immer noch die AIP auf Papier durch die Gegend und das nervt.... abgesehen vom Gewicht fange ich dann in einem Fall der Fälle an, hinten im Gepäckfach rumzukruschteln statt rauszugucken.
Anderenfalls hab ich am Panel ein Tablet, da alles drauf...  Ordner aufmachen... File vom Platz raussuchen, aufmachen, fertig.
Die Kollegen die die Anflugkarten georeferenziert in ihre Nav-Software einbauen wollen könnten das feature auch brauchen.
Viele Grüße!

**Comment by mipastgt** [comment]
So ganz verstehe ich Deinen Wunsch bzw. Deine Arbeitsweise noch nicht.

1. PDFs sind zum Ansehen und Durchblättern. Die darin enthaltenen Charts sind nicht georeferenziert und können daher auch nicht auf dem Nav-Display dargestellt werden.
2. Technisch ist es bereits möglich alle Seiten eines Flugplatzes als ein PDF zu exportieren. Das könnte man auch erweitern, so dass man dies für alle Flugplätze in einem gewählten Gebiet macht. Ist das der Wunsch?
3. Georeferenzierte Charts müssen in einem Format exportiert werden, dass die verwendete Nav-Software auch importieren kann. Besonders effizient geht das derzeit mit Enroute Flight Navigation, da damit alle georeferenzierten Charts für ein ganzes Gebiet auf einen Schlag importiert werden können. Andere Programme unterstützen bisher nur andere Formate und, soweit mir bekannt, geht das dann auch nur einzeln, was natürlich sehr mühsam ist.
4. Meine bisherige Vorgehensweise ist, dass man irgendwann seine Flugvorbereitung macht und sich dabei ein spezifisches Trip-Kit erstellt (natürlich inklusiver aller möglichen Alternativen) und dieses dann auf sein Mobilgerät lädt. Immer Gott und die Welt mitzunehmen führt doch nur zu einem unnötigen Platzverbrauch, erhöht den Transferaufwand, erhöht den Suchaufwand im Flug und führt u.U. dazu, dass man nicht die aktuellsten Informationen dabei hat. Die ganze AIP habe ich doch quasi per Internet immer dabei aber davon mache ich ja nur während der Flugvorbereitung gebraucht, oder nicht?

"Die Kollegen die die Anflugkarten georeferenziert in ihre Nav-Software einbauen wollen könnten das feature auch brauchen."
Das verstehe ich nicht. Was genau soll das bringen? Mit einem PDF können die doch nicht viel anfangen.

Konkret würde mich noch interessieren, ob ein Bedarf an einer Georeferenzierung der Roll-Charts besteht. Das würde extra Aufwand bedeuten, da diese Charts meist gedreht sind und keine Breiten- und Längengrade aufgedruckt sind. Technisch habe ich gerade heute verifiziert, dass Enroute damit klarkommen würde aber wie sieht es mit dem Bedarf aus? Um sinnvoll zu sein müssten die auch sehr genau georeferenziert sein damit ich nicht munter neben dem Rollweg entlangrolle.

**Reply by dorianiro** [reply]
Hallo,
2. ist der Wunsch.  Und das Gebiet wäre ganz DE.
Nichts georeferenzieren....  die Idee ist einfach, alle Karten aus der AIP dabeizuhaben und schnell drauf zugreifen zu können. NICHT in einer NAV-Anwendung, einfach pro Platz ein pdf, und mit einem pdf reader angucken.
AM BODEN  hast Du die AIP  per web immer dabei, das stimmt.  Aber nicht im Flug quer durch DE.  Du willst von Stettin nach Speyer und stellst unterwegs fest, daß es über dem Thüringer Wald doch eng werden könnte, abweichend von der Vorhersage. Dann fliegst Du eben ganz woanders hin, bist ja im Urlaub. Und freust dich über die Karte vom neuen Landeplatz.  
RICHTIG wichtig ist nicht. Denn die meisten NAV-Apps  (ich hab SD und VFRnav) zeigen ja standardmäßig die Platzrunden an, also runter kommt man schon. Ich hab trotzdem immer gern die Karten aus der AIP, zB wegen Bahnneigung und anderen Hinweisen.
...ich möchte aber auch nicht drängen. Dein Programm ist super, und schließlich kann man kaum jede exotische Idee umsetzen.
Danke und viele Grüße!

**Comment by mipastgt** [comment]
Keine Sorge, ich fühle mich nicht gedrängt :-) Für mich ist es nur wichtig die Wünsche der Anwender richtig zu verstehen und möglichst so zu koordinieren, dass es erstens nicht zu kompliziert wird und zweitens ein möglichst breiter Konsens bzgl. der Wünsche besteht. Technisch ist Dein so konkretisierter Wunsch nicht schwer umzusetzen. Ich schau mal was ich tun kann.

**Comment by dorianiro** [comment]
Ja suppi!
eins hatte ich noch übersehen, deine Frage wegen Geo Referenzierung. Mein Bekannter hatte das so gemacht, dass er zunächst die Karten von ganz DE einzeln runtergeladen hatte (würg). Und dann auf irgendeiner Webseite, die ich jetzt nicht kenne, hochgeladen, dort georeferenziert, und das Ergebnis dann im richtigen Ordner bei seiner NAV-SW abgelegt.
Deswegen meine Bemerkung: wer das auch so machen will, der freut sich bestimmt auch über einen Massendownload.
... und bitte entschuldige mein blöden Nicknamen hier. Alle möglichen Namen waren schon belegt, dann hatte ich einfach wild rumprobiert.
Im UL-Forum bin ich Maraio.
Grüße!

**Comment by mipastgt** [comment]
Mit der neuen Version 1.1.3 sollte das meiste davon jetzt gehen.

**Reply by dorianiro** [reply]
Der Bericht!
Ich hab 1.1.3 installiert und finde nix dazu, auch nicht in der Hilfe.  Geht also vermutlich als tripkit.   OK, unter den Beispielen hab ich eins gefunden "flying in germany"  (evtl die Beispiele noch beschriften, da steht nur 4x 'beispiel' als linktext)
Runtergeladen, select/export.
Das system fragt garnicht wohin ich speichern will?!   Am Schluß kommt ne fehlermeldung, Bild 99.  Aha, da steht auch wohin gespeichert werden soll.  WÜRG, in den Windoof Standard-Userordner! In Deiner Anleitung unter 4.3.3. steht das ja auch, man muß nur lesen...  (meinen usern in der Firma sage ich, daß der Standard-Benutzerordner Böse ist... die sollen temporären Müll auf der lokalen Datenpartition speichern und wenns kein Müll ist dann BITTE auf dem server im Userordner!)
Grins.
Ok, ich gucke rein und finde viele webps.  Dann lese ich die tripkit-Hilfe. Und stelle um von webp auf jpg, und addiere  pageFormat | pdf |   und asSingleFile | false

Und lasse das tripkit wieder laufen.
1. keine Fehlermeldung am Ende
2. es wird ein ZIP erzeugt und gespeichert.
Ich packe das ZIP aus, viele JPGs, allerdings wieder nur die Anflugkarten, nicht das Flugplatz-Layout='Rollkarte'.  
Leider kein einziges pdf.
So hatte ich das json ergänzt:
       "minInvScale": 50000.0,
        "maxInvScale": 90000.0,
        "imageFormat": "jpg"
	"pageFormat": "pdf"
	"asSingleFile": false

Bestimmt siehst Du den Fehler.  Und: ein 'fliegender Steuerberater'  ist hier bestimmt überfordert....  :-)
Danke und viele Grüße!  Volkmar
![ScreenShot0099](https://github.com/mipastgt/AIPBrowserDE-Public/assets/147976567/e8387595-ac34-4cba-9dba-56e289973e3e)

**Reply by mipastgt** [reply]
1. Beispiele sind jetzt beschriftet. (version 1.1.3-v3, 2023-12-01)
2. Erst lesen - dann machen :-)
3. Wie bereits an anderer Stelle geschrieben, gibt es die Flugplatz-Layout='Rollkarte' nur als ganze PDF-Seiten der AIP.
4. Hilfe zur Trip-Kit-Konfiguration kann ich nur geben, wenn ich die ganze Datei habe (hier als Anhang einfügen) und nicht nur ein paar Fragmente.

**Reply by dorianiro** [reply]
jetzt funzts. Man muß eine extra Sektion für pages aufmachen, auch mit 'covered area'. Siehe Deine Beispiele.... das hatte ich nicht beachtet. Saugut!  DANKE!
Solltest Du mal ein Flugfunk-HFG von Yaesu oder das neue von Wouxun ('gut und günstig') brauchen, mach ich Dir gern nen Spezialpreis!

**Comment by mipastgt** [comment]
Dann bin ich ja froh, dass jetzt alles funktioniert :-)

**Reply by dorianiro** [reply]
und ich erst...   :-)

**Comment by mipastgt** [comment]
Ich habe in der letzten Zeit auch häufiger Trip-Kits für die ganze deutsche AIP erstellt. Insbesondere in Kombination mit der neuen Unterstützung für [ForeFlight Content Packs](https://foreflight.com/support/content-packs/) geht das jetzt super. (Für das bisherige native Format geht das genauso aber da fehlt dann halt der Komfort in der Anwendung.)

**Comment by dorianiro** [comment]
Ähm… die Nachricht war jetzt für alle und nicht speziell für mich, oder?

 

Aktuell bin ich noch im Winterschlaf was weitere Flüge angeht.
Guck ich mir an wenn es wieder losgeht. Danke!

 

Viele Grüsse

Volkmar 

AeC Landau

 

 

Von: Michael Paus ***@***.***> 
Gesendet: Donnerstag, 25. Januar 2024 10:57
An: mipastgt/AIPBrowserDE-Public ***@***.***>
Cc: dorianiro ***@***.***>; Author ***@***.***>
Betreff: Re: [mipastgt/AIPBrowserDE-Public] Anflugkarten bulk download (Discussion #4)

 

Ich habe in der letzten Zeit auch häufiger Trip-Kits für die ganze deutsche AIP erstellt. Insbesondere in Kombination mit der neuen Unterstützung für ForeFlight Content Packs <https://foreflight.com/support/content-packs/>  geht das jetzt super. (Für das bisherige native Format geht das genauso aber da fehlt dann halt der Komfort in der Anwendung.)

**Reply by mipastgt** [reply]
Für Dich und alle die es sonst noch interessieren könnte. :-)

**Comment by dorianiro** [comment]
😊 !

 

Danke

V

 

 

Von: Michael Paus ***@***.***> 
Gesendet: Montag, 5. Februar 2024 23:02
An: mipastgt/AIPBrowserDE-Public ***@***.***>
Cc: dorianiro ***@***.***>; Author ***@***.***>
Betreff: Re: [mipastgt/AIPBrowserDE-Public] Anflugkarten bulk download (Discussion #4)

 

Für Dich und alle die es sonst noch interessieren könnte. :-)
